File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (a) {
    let count = 0;
    let mp = new Map()



    for (let i = 0; i < a.length; i++) {
        if (mp.has(a[i])) {
            mp.set(a[i], mp.get(a[i]) + 1)
        }

        if (!mp.has(a[i])) {
            mp.set(a[i], 1)
        }

    }

    for (const [key, value] of mp) {
        if (value > a.length / 2) {
            count = key
        }
    }

    return count;
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int n = nums.length;

        for(int num : nums){
            mp.put(num, mp.getOrDefault(num, 0) + 1);
        }

        int res = 0;
        for(var it : mp.entrySet()){
            if(it.getValue() > n / 2){
               res = it.getKey();
            }
        }

        return res;

    }
}